dataland-backend/src/main/kotlin/org/dataland/datalandbackend/api/DataOwnerApi.kt

```diff
@@ -0,0 +1,129 @@
+package org.dataland.datalandbackend.api
+
+import io.swagger.v3.oas.annotations.Operation
+import io.swagger.v3.oas.annotations.responses.ApiResponse
+import io.swagger.v3.oas.annotations.responses.ApiResponses
```

Comment:
![image](https://github.com/d-fine/Dataland/assets/104447806/585698b9-7ee2-463f-8490-8447dd91b0ac)
The marked error should be discussed. If the company does not exist on dataland i would expect to get a 404 error.

Reply:
why exactly? i mean it's relevant to know if the company doesnt exist at all or if it justst doesn't have the specified owner

Reply:
Yes, but that could be done using the error message. Error code 400 is a Bad Request error. From my point of view a non existing company is not a bad request to the server, but rather a 404 ressource not found.

Reply:
so you want to kind of have two different 404 responses?

Reply:
Would make sense from my point of view

Reply:
this is the case now

Reply:
ok, closing this now